app/views/works/_work_activity_provenance.html.erb

```diff
@@ -4,12 +4,15 @@
     No activity
   <% end %>
   <ul class="beads">
-    <% @work_decorator.changes.sort_by(&:created_at).each do |activity| %>
+    <% @work_decorator.changes.take(100).each_with_index do |activity| %>
```

Comment:
This was culprit number 2. The code to display each notification issues a SQL SELECT to get the user information, even with Rails using its internal cache 1000 individual SQL SELECTS are slow. For now we are displaying only the first 100 notifications (if there are more they can contact us and we'll look closer)